scripts/lighthouse_checks.py

```diff
@@ -0,0 +1,172 @@
+    """Download and install GO."""
+    outfile_name = 'go-download'
+
+    if common.is_windows_os():
+        pass
+    else:
+        extension = '.tar.gz'
+
+        if common.is_mac_os():
+            go_file_name = 'go%s.darwin-amd64' % (GO_VERSION)
+        elif common.is_linux_os():
+            go_file_name = 'go%s.linux-amd64' % (GO_VERSION)
+
+        url_to_retrieve = 'https://storage.googleapis.com/golang/%s%s' % (
+            go_file_name, extension)
+
+        python_utils.url_retrieve(url_to_retrieve, outfile_name)
+
+        subprocess.check_call([
+            'tar', '-xvzf', outfile_name, '-C', common.OPPIA_TOOLS_DIR
+        ])
+
+    os.rename(
+        os.path.join(common.OPPIA_TOOLS_DIR, 'go'),
+        GO_PATH)
+    python_utils.PRINT('Installation successful')
+
+
+def download_and_install_compy():
+    """Download and install the compy proxy server."""
+
+    subprocess.check_call([GO_BINARY, 'get', 'github.com/barnacs/compy'])
+    with common.CD(COMPY_DOWNLOAD_PATH):
+        subprocess.check_call([GO_BINARY, 'install'])
+    python_utils.PRINT('Installation successful')
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path),
+        '.*%s.*' % re.escape(COMPY_BINARY)
+    ]
+
+    for p in RUNNING_PROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+
+def delete_reports():
+    """"Delete the reports that are stored in the lighthouse ci folder."""
+    bash_command = 'rm -r .lighthouseci'
+    process = subprocess.Popen(bash_command.split(), stdout=subprocess.PIPE)
+    process.communicate()
+
+
+def start_proxy_server():
+    """Start compy proxy server to serve gzipped assets."""
+    ssl_options = '-cert cert.crt -key cert.key -ca ca.crt -cakey ca.key'
+    p = subprocess.Popen([COMPY_BINARY, '-host', ':%s' % COMPY_SERVER_PORT,
+                          '-gzip', '9', ssl_options])
+    RUNNING_PROCESSES.append(p)
+
+
+def wait_for_port_to_be_open(port_number):
+    """Wait until the port is open.
+    Args:
+        port_number: int. The port number to wait.
+    """
+    waited_seconds = 0
+    while (not common.is_port_open(port_number) and
+           waited_seconds < MAX_WAIT_TIME_FOR_PORT_TO_OPEN_SECS):
+        time.sleep(10)
+        waited_seconds += 2
+    if (waited_seconds ==
+            MAX_WAIT_TIME_FOR_PORT_TO_OPEN_SECS and
+            not common.is_port_open(port_number)):
+        python_utils.PRINT(
+            'Failed to start server on port %s, exiting ...' % port_number)
+        sys.exit(1)
+
+
+def run_lighthouse_checks():
+    """Runs the lighthouserc.js config with bash command lhci autorun."""
+    bash_command = 'lhci autorun'
+    process = subprocess.Popen(bash_command, shell=True, stdout=subprocess.PIPE)
```

Comment:
Do not do it this way, use the binary file of `lhci` directly as in [run_frontend_tests.py](https://github.com/oppia/oppia/blob/develop/scripts/run_frontend_tests.py#L88) with `karma`.

Reply:
I'm not sure if there is a binary file for @lhci, at least I'm having some trouble finding it. I did find an ⁨autorun.js in node_modules⁩/@lhci⁩/⁨cli⁩/src⁩/⁨autorun⁩ , however. I'm curious why we shouldn't use the builtin lhci autorun command?

Reply:
`lhci autorun` might not be available everywhere, and in most of the codebase we access the binary/js files directly. There should be either binary file or you should be able to run the `.js` file using the node binary file (as in [html_linter.py](https://github.com/oppia/oppia/blob/develop/scripts/linters/html_linter.py#L356))

Reply:
Hmm in the documentation, it shows that lhci autorun command is available once you install lhci, but I guess it makes sense to follow what the rest of the codebase does. 

https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/cli.md

Do you know what the `'--rc = htmllintc' `arg is for in html_linter.py

```
        node_path = os.path.join(common.NODE_PATH, 'bin', 'node')
        htmllint_path = os.path.join(
            'node_modules', 'htmllint-cli', 'bin', 'cli.js')

        error_summary = []
        total_error_count = 0
        summary_messages = []
        stdout = sys.stdout
        htmllint_cmd_args = [node_path, htmllint_path, '--rc=.htmllintrc']
```

Reply:
You don't need to care about that, just the part how node and htmllint paths work here.

Reply:
> Hmm in the documentation, it shows that lhci autorun command is available once you install lhci

I think it will be available only if you run it from the npm context or if you install it globally.
There should be a binary for it at `node_modules/.bin`

Reply:
Done